src/smartcontracts/nativeSerializer.ts

```diff
@@ -333,6 +337,16 @@ export namespace NativeSerializer {
         errorContext.throwError(`(function: toEnumValue) unsupported native type ${typeof native}`);
     }
 
+    function toManagedDecimal(native: any, type: ManagedDecimalType, errorContext: ArgumentErrorContext): TypedValue {
+        if (typeof native === "object") {
+            if (type.getMetadata() == "usize") {
+                return new ManagedDecimalValue(native[0], native[1], true);
+            }
+            return new ManagedDecimalValue(native[0], native[1]);
```

Comment:
Can also call constructor once, and always pass the last parameter (whether it's false or true).